### PR TITLE
[#101] eav:attribute:view command ported

### PR DESCRIPTION
**Pull request by robbieaverill**
* Added abstract attribute command and test
* Added ViewCommand and test
* Logic refactored from Magerun 1 equivalent a little

Covers #101

### CODE REVIEW

**Comment by ktomk** [comment]
@robbieaverill: I could only quickly review the code and have left two smaller comments on some details and a more elaborated one on the abstract test-case with another tip for organizing the test methods and how to name them. Please take a look and share what you think.

**Comment by robbieaverill** [comment]
Hey Tom,

My laptop died while writing a reply to you. I have updated a few elements from your comments, they are pushed now. I will send you my reply on Tuesday when I have a laptop charger :-)

Cheers
Robbie

**Comment by robbieaverill** [comment]
Hey @ktomk 

Thanks again for the code review. I'll just add a couple of comments to your suggestions:

> In PHP FQCNs do not start with the backslash (\), please do not add it in front. It's especially not needed in string parameters as those are always fully qualified

No problem - I had copied from [the doc block of the Magento class](https://github.com/magento/magento2/blob/develop/app/code/Magento/Eav/Model/Config.php#L421), but have removed the trailing slash throughout now.

> I think you did this as well with the previous PR: I'd normally say that it's not neccesary to put the abstract class under test (and here in PHP you even need to use the mock-builder to do that), it's much more straight forward to test that method on the concrete ViewCommand for which you also added the test.

I agree and disagree - agree because the reset of the unit tests are all specific to commands, so these kinds of abstract tests seem a little out of place. I don't believe they *are* out of place, as I believe that their methods should be tested if they can be.

I disagree with your statement because the `getAttribute()` method is not part of the ViewCommand. It's part of the abstract class, so it should be tested as part of it. If we were to remove the ViewCommand and its test we may then lose the test for the `getAttribute()` method which belongs to the abstract class... you know.

>This shortens the name and removed redundancy. Also you don't test what should, you just test ;) - Here the attribute model. This also helps to reduce words and make the language used more straight forward.... In this example the test method is named as the method under test

I imagine that this mostly comes down to your preference of approach. Aside from the test prefix which isn't really a big deal either way, I personally prefer the naming of test methods for what the test is supposed to do. [See reference](https://jtreminio.com/2013/03/unit-testing-tutorial-introduction-to-phpunit/#method-test-names) and [another reference](http://stackoverflow.com/a/4656282/2812842) - it especially helps when getting reports of test coverage, as you can see what the unit tests are supposed to be ("should ...") testing for.

An important thing to note here is that the style of these unit tests and the content of the majority of the ones in this project so far are not so much unit tests as integration tests, seeing as they are actually testing the Command's compatibility with the Symfony console, then the ability for it to execute correctly, then the ability to gather information and process it as expected, then the ability to output a formatted result somehow. In this instance, it's probably OK to name an integration test after the method (such as `execute()`) that controls all of that logic, however I still believe that unit tests should be named after the expected behaviour of the method you're testing (just me!).

## TL;DR

At the end of the day, it's your project - I'm more than happy to do things your way, just thought I should explain why I've done things this way :)

Cheers,
Robbie

**Comment by robbieaverill** [comment]
Sorry for the commit spam - have updated again to fix a couple of formatting issues raised by Scrutinizer:

* Line length exceeded
* Spacing between `(array)` cast and the variable

**Comment by ktomk** [comment]
Hi @robbieaverill, thanks for your extensive feedback! It's not so much the questions who's project that is (and it's also yours or at least I hope so!), but in my eyes the exchange. Your criticism picks some good points, for example with the unit tests vs. integration tests. It has much that in my eyes is even more important than a method name.

The name is what the test is called.

As simple as this looks like, as hard this is to achieve. And as much this can be discussed. But perhaps better with a drink or two than all the times in PR comments. If it's OK for you, I just leave some comments here or there on the code and you pick what you think is useful and for the rest just leave it where it is. No hard feelings and far away from obligatory what I write for the concrete PR. The opposite is the case: Thanks for writing so many tests! I hope the review tools set-up are useful (e.g. the feedback by Scrutinizer) and if you see anything fishy, just raise it. We don't have strict rules and you should feel welcome.

And there is no commit spam :)

**Comment by robbieaverill** [comment]
@ktomk yet ;)

Thanks for the message!

Cheers,
Robbie
